### PR TITLE
[Issue #1820] Refactor care plan contacts controller spec to request spec

### LINKED ISSUE #1820: Refactor spec/controllers/care_plan_contacts_controller_spec.rb to spec/request/care_plan_contact_spec.rb
# Description

Controller Specs had been obsolete since Rails 5. It is best practice to create request spec instead of controller spec as recommended by the Rspec team.

Recommended Reading:

- [Stackoverflow Discussion in regards to Controller Specs vs Request Specs](https://stackoverflow.com/questions/40851705/controller-specs-vs-request-specs)
- [Medium article discussing request and controller specs](https://medium.com/just-tech/rspec-controller-or-request-specs-d93ef563ef11)

Resources on Request Spec formatting:

- [RelishApp](https://relishapp.com/rspec/rspec-rails/v/4-0/docs/request-specs/request-spec)

Original Issue: https://github.com/ifmeorg/ifme/issues/1815

---

Please assign yourself (via the Assignees dropdown), if you do want to work on this issue. Can't find yourself? You need to [join our organization](https://github.com/ifmeorg/ifme/wiki/Join-Our-Slack).

Check out our [Picking Up Issues](https://github.com/ifmeorg/ifme/wiki/Picking-Up-Issues) guide if you haven't already!

### PR DESCRIPTION
**Pull request by camilacampos**
# Description

This refactors the tests associated with the CarePlanContactsController to use RSpec request type tests instead of controller

## Corresponding Issue

Fixes #1820 
Parent issue: https://github.com/ifmeorg/ifme/issues/1815

---

Reviewing this pull request? Check out our [Code Review Practices](https://github.com/ifmeorg/ifme/wiki/Code-Review-Practices) guide if you haven't already!

### CODE REVIEW

**Comment by tyler-wel** [comment]
Just quick comment, I'd recommend reading the contributing wiki
https://github.com/ifmeorg/ifme/blob/main/CONTRIBUTING.md
https://github.com/ifmeorg/ifme/wiki/Picking-Up-Issues

While it is great for you to contribute so much, the if-me maintainers ask you assign only *one* issue at a time to yourself, giving others the opportunity to contribute also

File: spec/requests/care_plan_contacts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+  describe '#create' do
+    context 'when the user is logged in' do
+      before { sign_in user }
+
+      context 'when valid params are supplied' do
+        it 'creates the care plan contact' do
+          valid_params = { name: 'Person Name', phone: '416000000' }
+
+          post care_plan_contacts_create_path, params: { care_plan_contact: valid_params }
```
Comment (tyler-wel) [inline spec/requests/care_plan_contacts_spec.rb:14]: is `valid_params` used anywhere is else? if not, I'd say you can refactor into one line.

Don't really _have_ to make everything into it's own variable
```suggestion
          post care_plan_contacts_create_path, params: { 
            care_plan_contact:  { 
              name: 'Person Name', phone: '416000000' 
            } 
          }
```

File: spec/requests/care_plan_contacts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      end
+    end
+
+    context 'when the user is not logged in' do
+      it 'redirects to sign_in page' do
+        valid_params = { name: 'Person Name', phone: '416000000' }
+
+        post care_plan_contacts_create_path, params: { care_plan_contact: valid_params }
+
+        expect(response).to redirect_to new_user_session_path
```
Comment (tyler-wel) [inline spec/requests/care_plan_contacts_spec.rb:42]: @bennpham are we keeping `redirect_to` ?

**Comment by camilacampos** [comment]
@tyler-wel, thanks for the review! Sorry for picking more than 1 issue, I have some time available and got carried away coding  🤣 I promisse it won't happen again.

File: spec/requests/care_plan_contacts_spec.rb (hunk 2 above)
Comment (camilacampos) [reply spec/requests/care_plan_contacts_spec.rb:46]: if not, there are some other `redirect_to`s on request specs.

**Comment by tyler-wel** [comment]
> @tyler-wel, thanks for the review! Sorry for picking more than 1 issue, I have some time available and got carried away coding 🤣 I promisse it won't happen again.

Nah it's no big deal 😄 just with hacktober, probably good idea to follow guideline and give chance to everyone 👍

File: spec/requests/care_plan_contacts_spec.rb (hunk 2 above)
Comment (bennpham) [reply spec/requests/care_plan_contacts_spec.rb:46]: @tyler-wel @camilacampos `redirect_to` seems fine when I removed `rails-controller-testing` gem so I don't have a problem with it.

**Review by bennpham — APPROVED** [review]
This looks good to me 👍

**Pull request merged**